toolkit/tools/internal/safechroot/safechroot.go

```diff
@@ -689,3 +694,33 @@ func (c *Chroot) GetMountPoints() []*MountPoint {
 	mountPoints := append([]*MountPoint(nil), c.mountPoints...)
 	return mountPoints
 }
+
+// stopGPGAgent stops gpg-agent and keyboxd if they are running inside the chroot.
+//
+// It is possible that one of the packages or post-install scripts started a GPG agent.
+// e.g. when installing the azurelinux-repos SPEC, a GPG import occurs. This starts the gpg-agent process inside the chroot.
+// To be able to cleanly exit the setup chroot, we must stop it.
+func (c *Chroot) stopGPGAgent() {
+	_, err := exec.LookPath("gpgconf")
+	if err != nil {
+		// gpgconf is not installed, so gpg-agent is not running.
+		logger.Log.Warnf("gpgconf is not installed, so gpg-agent is not running: %s", err)
```

Comment:
I'd suggest either an info or debug level log - this is expected behaviour. Warnings in cases like these tend to lead to confusion for the users and during debugging attempts.

Reply:
Fixed!